ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionMapping.java

```diff
@@ -33,16 +34,38 @@ public IngestionMapping(String ingestionMappingReference, IngestionMappingKind i
     }
 
     /**
-     * Sets the ingestion mapping parameters
+     * Please use setIngestionMappingReference for production as passing the mapping every time is wasteful
+     * @param ingestionMapping          Array of columnMappings of the same kind.
+     * @param ingestionMappingKind      IngestionMappingKind: the format of the source data to map from.
+     */
+    public IngestionMapping(ColumnMapping[] ingestionMapping, IngestionMappingKind ingestionMappingKind) {
+        this.ingestionMapping = ingestionMapping;
+        this.ingestionMappingKind = ingestionMappingKind;
+    }
+
+    /**
+     * Sets the ingestion mapping reference parameters
      *
      * @param ingestionMappingReference String: the name of the pre-defined ingestion mapping.
      * @param ingestionMappingKind      IngestionMappingKind: the format of the source data to map from.
      */
-    public void setIngestionMapping(String ingestionMappingReference, IngestionMappingKind ingestionMappingKind) {
+    public void setIngestionMappingReference(String ingestionMappingReference, IngestionMappingKind ingestionMappingKind) {
```

Comment:
I'm not sure I would mix Mapping reference into this class. but rather have a separate one instead as it makes it unclear what are mandatory fields and what aren't (also would not allow setters , just setters to enforce correct usage)

Reply:
again just wanted to do same things as in the .Net - don't think this should be different here ..

Reply:
Vladik agreed that this should be consistent with the .Net